### ISSUE TITLE
Difference in interfaces; i.e. accessing through xgboost.train(...) vs clf = XGBClassifier(...); clf.fit(...)

### ISSUE THREAD

**Issue by thusithaC** [dmlc/xgboost#1099]
Hi,

Is it correct to say that there are 2 interfaces for xgboost in python? For example, the documentation in http://xgboost.readthedocs.org/en/latest/python/python_api.html uses xgboost.train( )with Dmatrices as input. As an alternative I've seen people use,  clf = XGBClassifier(...); clf.fit(); 

The problem I have is there are few parameters that look different. For example, xgboost.train( ) interfaces has an 'eta', whereas  clf = XGBClassifier(...) has an learning rate. Are these two comparable? 

Furthermore clf = XGBClassifier(...) has a 'scale_pos_weight' param, and it doesnt seem to work with  xgboost.train( ). I heard this is useful when the datasets are severely imbalanced. Could someone clarify this issue? 

So to summarize, are the two interfaces the exact same, if so why some params have different names, and some seems missing altogether? 

Thanks, 
T

**Comment by tqchen** [comment]
xgb.train is the original training interface that works with all the parameters, including ```scale_pos_weight```, whichi can be passed as parmater.

The scikit-learn style interface changed a few arguments to be consistent with scikit-learn. But most parameters should remain consistent
